packages/extension-api/src/extension-api.d.ts

```diff
@@ -1429,6 +1429,15 @@ declare module '@podman-desktop/api' {
 
   type NotificationType = 'info' | 'warn' | 'error';
 
+  export interface NotificationTask extends Task {
```

Comment:
Why do we want to expose these interfaces (this one and also the other added ones) to the extensions API? If I place myself as a user of the extension, I really don't understand what to do with these interfaces, as they are referenced in no function of the API

Reply:
Because they will be needed for https://github.com/containers/podman-desktop/issues/7709

When using `showNotification` you will get a NotificationTask instead of just a Disposable 

https://github.com/containers/podman-desktop/blob/86c7ee8b7dd09225a0b59dcfcf535021664bc48f/packages/extension-api/src/extension-api.d.ts#L1962

We will also expose a new `createTask` method similar to `showNotification` to let the extension do more stuff with the notifications/tasks

Reply:
I don't see yet why Tasks should leak into the extension API

it also still need to return a Disposable for showNotification (you can't change as it'll break API)

Reply:
NotificationTask extend Task which extend Disposable, it would not break the API

Reply:
I moved the Task and NotificationTask interface to the main.